### PR TITLE
HDDS-9242. Merge unit and integration checks into a combined one

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Unit and integration tests are currently run in separate checks.  Integration tests are split into parallel runs to reduce CI wall clock time.  Unit tests, however, are run in one check, which is now one of the slowest ones.

This PR changes `integration` check to also execute unit tests, to benefit from running them in parallel splits.

It also rebalances splits by creating some new ones and merging `contract` into `filesystem`.

Some tests related to snapshots are moved from `org.apache.hadoop.ozone.om` to `org.apache.hadoop.ozone.om.snapshot` to reduce the time required for `integration (om)` check.

https://issues.apache.org/jira/browse/HDDS-9242

## How was this patch tested?

Checked CI output to verify total number of tests across all splits is about the same as previously.

CI:
https://github.com/adoroszlai/ozone/actions/runs/7829598697

### CODE REVIEW

File: dev-support/ci/selective_ci_checks.bats [hunk 1]
```diff
@@ -195,7 +195,7 @@ load bats-assert/load.bash
 #@test "native test in other module" {
 #  run dev-support/ci/selective_ci_checks.sh 7d01cc14a6
 #
-#  assert_output -p 'basic-checks=["rat","author","checkstyle","findbugs","native","unit"]'
+#  assert_output -p 'basic-checks=["rat","author","checkstyle","findbugs","native"]'
```
Comment (Galsza) [inline dev-support/ci/selective_ci_checks.bats:198]: This is commented out but the `assert_output -p needs-integration-tests=false` stayed the same leaving it in an inconsistent state. Are there any reason to reenable this in the future?

**Review by Galsza — APPROVED** [review]
@adoroszlai thank you for this patch, it looks good to me. I've left a minor comment regarding a commented out test.

**Comment by Galsza** [comment]
LGTM+1

**Review by errose28 — COMMENTED** [review]
Thanks for working on this @adoroszlai. Just had a few questions since I'm not the most familiar with some of these CI shell scripts.

File: dev-support/ci/selective_ci_checks.bats [hunk 2]
```diff
@@ -189,20 +189,17 @@ load bats-assert/load.bash
...
-#  assert_output -p 'basic-checks=["rat","author","checkstyle","findbugs","native","unit"]'
-#  assert_output -p needs-build=true
-#  assert_output -p needs-compile=true
-#  assert_output -p needs-compose-tests=false
-#  assert_output -p needs-dependency-check=false
-#  assert_output -p needs-integration-tests=false
-#  assert_output -p needs-kubernetes-tests=false
-#}
+@test "native test in other module" {
+  run dev-support/ci/selective_ci_checks.sh 822c0dee1a
```
[the remark this answers is not in the document]
Comment (adoroszlai) [reply dev-support/ci/selective_ci_checks.bats:193]: This test case verifies that changing a test tagged as `@Native` but outside of the `hdds-rocks-native` module triggers the `unit (native)` check.  Previously it used:

```
commit 7d01cc14a69cfbad9f5bee3b4f7da0a67c202711

    HDDS-8834. Ignore TestManagedSstFileReader#testGetKeyStreamWithTombstone (#4881)

 hadoop-hdds/rocksdb-checkpoint-differ/src/test/java/org/apache/ozone/rocksdb/util/TestManagedSstFileReader.java | 2 ++
 1 file changed, 2 insertions(+)
```

but ddc19e9c1d renamed `TestManagedSstFileReader` to `TestSstFileSetReader`.  When `selective_checks.sh` looks for `@Native` tests, it no longer finds `TestManagedSstFileReader`, so it does not trigger `unit (native)` for this specific test class.  At the time of ddc19e9c1d I didn't find any other commit that could be used as a test case, so I had to disable it.  Now there is such a commit:

```
commit 822c0dee1a8fd35ba2a22470e73ebd68b295edd3

    HDDS-10148. TestOmSnapshotFsoWithNativeLib should be tagged as native test (#6160)

 hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestOmSnapshotFsoWithNativeLib.java | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)
```

This part of the change could be done separately, but I think it's a very small item that fits here.

File: hadoop-ozone/dev-support/checks/unit.sh [hunk 3]
```diff
@@ -14,8 +14,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-#checks:unit
-
```
[the remark this answers is not in the document]
Comment (adoroszlai) [reply hadoop-ozone/dev-support/checks/unit.sh:18]: We no longer want to run `unit.sh` as part of CI.  All non-native JUnit tests are run by `integration.sh` due to the removal of project filter.

`unit.sh` is kept for developer use.

[the remark this answers is not in the document]
Comment (adoroszlai) [reply hadoop-ozone/dev-support/checks/unit.sh:18]: We should probably restore `native` as a top-level check (rename `unit` to `native`, remove matrix strategy, and simplify related logic in scripts).  Filed HDDS-10598 for that.

**Review by errose28 — APPROVED** [review]
Thanks @adoroszlai LGTM

**Comment by adoroszlai** [comment]
Thanks @errose28, @Galsza for the review.

**Pull request merged**